service/src/main/java/bio/terra/landingzone/stairway/flight/ResourceNameRequirements.java

```diff
@@ -0,0 +1,3 @@
+package bio.terra.landingzone.stairway.flight;
+
+public record ResourceNameRequirements(String resource, int maxNameLength) {}
```

Comment:
I think the `resource` variable would be better named `resourceType` as this is expressing the relationship between a resource _type_ and a length constraint, not a specific resource instance.